http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClientWireMockTest.java

```diff
@@ -558,6 +563,24 @@ public ChannelFuture close() {
         assertThat(channelClosedFuture.get(5, TimeUnit.SECONDS)).isTrue();
     }
 
+    @Test
+    public void execute_requestByteBufferWithNonZeroPosition_shouldHonor() throws Exception {
+        String body = randomAlphabetic(70);
+        byte[] content = randomAscii(100).getBytes();
+        ByteBuffer requestContent = ByteBuffer.wrap(content);
+        requestContent.position(95);
+        String expected = new String(content, 95, 5);
+
+        URI uri = URI.create("http://localhost:" + mockServer.port());
+        stubFor(any(urlPathEqualTo("/"))
+                    .withRequestBody(equalTo(expected)).willReturn(aResponse().withBody(body)));
+        SdkHttpRequest request = createRequest(uri, "/", expected, SdkHttpMethod.POST, emptyMap());
+        RecordingResponseHandler recorder = new RecordingResponseHandler();
+
+        client.execute(AsyncExecuteRequest.builder().request(request).requestContentPublisher(new SimpleHttpContentPublisher(requestContent)).responseHandler(recorder).build());
+        recorder.completeFuture.get(5, TimeUnit.SECONDS);
```

Comment:
Shouldn't there be some additional assertions here?

Reply:
The request would fail without the change, but yeah, I can add additional wiremock verification.